onnxruntime/core/providers/nnapi/nnapi_builtin/builders/helper.h

```diff
@@ -146,5 +155,9 @@ std::string Shape2String(const std::vector<uint32_t>& shape);
 bool CheckIsInitializer(const InitializedTensorSet& initializers, const Node& node,
                         size_t index, const char* input_name) ORT_MUST_USE_RESULT;
 
+inline std::unordered_map<const Node*, std::optional<QDQ::NodeGroup>> target_node_to_qdq_group;
```

Comment:
do we need global variables? could these be parameters to the helper functions instead?

Reply:
Instead of having standalone functions and a vector of Nodes in OpSupportChecker can we have a helper class that wraps up all the QDQ functionality and data, and an instance of that class in OpSupportChecker? That way the ownership semantics are clearer.